mbedtls/src/ssl/context.rs

```diff
@@ -172,35 +177,77 @@ impl<T> Context<T> {
     }
 }
 
+/// # Safety
+/// `io` must live as long as `ctx` or the next time bio is set/cleared.
+unsafe fn set_bio_raw<IoType, T: IoCallbackUnsafe<IoType>>(ctx: *mut ssl_context, io: &mut T) {
+    ssl_set_bio(
+        ctx,
+        io as *mut T as *mut c_void,
+        Some(T::call_send),
+        Some(T::call_recv),
+        None,
+    );
+}
+
+#[cfg(all(feature = "std", feature = "async"))]
+impl<T> Context<T>  {
+    pub(super) fn with_bio_async<'cx, R, IoType>(&mut self, cx: &mut std::task::Context<'cx>, f: impl FnOnce(&mut Self) -> R) -> Option<R> where for<'c> (&'c mut std::task::Context<'cx>, &'c mut T): IoCallbackUnsafe<IoType> {
```

Comment:
This must be a TLS thing, but I feel like somewhere there should be a clue as to what "BIO" stands for in one of the comments... or is it just like, buffered-I/O?

If it's just buffered I/O, then why not ``with_buf_io_async``?

Reply:
`bio` is a concept in common TLS implementation which refers to basic IO.
openssl and mbedtls both use this concept: 
Ref: https://stackoverflow.com/questions/51672133/what-are-openssl-bios-how-do-they-work-how-are-bios-used-in-openssl

I will update comments for explaining this